Pod/Classes/XExtensionItem.m

```diff
@@ -44,38 +61,111 @@ - (id)activityViewControllerPlaceholderItem:(UIActivityViewController *)activity
     return self.placeholderItem;
 }
 
+- (NSString *)activityViewController:(UIActivityViewController *)activityViewController dataTypeIdentifierForActivityType:(NSString *)activityType {
+    return self.dataTypeIdentifier;
+}
+
 - (id)activityViewController:(UIActivityViewController *)activityViewController itemForActivityType:(NSString *)activityType {
-    NSExtensionItem *item = [[NSExtensionItem alloc] init];
-    item.userInfo = ({
-        NSMutableDictionary *mutableUserInfo = [[NSMutableDictionary alloc] init];
-        [mutableUserInfo addEntriesFromDictionary:self.userInfo];
-        
-        NSMutableDictionary *mutableParameters = [[NSMutableDictionary alloc] init];
-        [mutableParameters setValue:self.tags forKey:ParameterKeyTags];
-        [mutableParameters setValue:self.sourceURL forKey:ParameterKeySourceURL];
-        [mutableParameters addEntriesFromDictionary:self.referrer.dictionaryRepresentation];
-        
-        if ([mutableParameters count] > 0) {
-            mutableUserInfo[ParameterKeyXExtensionItem] = [mutableParameters copy];
-        }
-        
-        mutableUserInfo;
-    });
-    
     /*
-     The `userInfo` setter *must* be called before the following three setters, which merely provide syntactic sugar for
-     populating the `userInfo` dictionary with the following keys:
-     
-     * `NSExtensionItemAttributedTitleKey`,
-     * `NSExtensionItemAttributedContentTextKey`
-     * `NSExtensionItemAttachmentsKey`.
+     Share extensions take `NSExtensionItem` instances as input, and *some* system activities do as well, but some do
+     not. Unfortunately we need to maintain a hardcoded list of which system activities we can pass extension items to.
      
+     Trying to pass an extension item into a system activity that doesn’t know how to process it will result in no data 
+     making it’s way through. In these cases, we’ll pass the placeholder item that this instance was initialized with 
+     instead.
      */
-    item.attachments = self.attachments;
-    item.attributedTitle = self.attributedTitle;
-    item.attributedContentText = self.attributedContentText;
+    if (activityTypeAcceptsExtensionItemInput(activityType)) {
```

Comment:
This function reads oddly as it seems to suggest I’d be passing in a `ExtensionItemInput` but instead I’m passing in an activity type.